modules/core/src/test/java/org/apache/ignite/internal/processors/closure/ClusterMetricsOnCacheSelfTest.java

```diff
@@ -0,0 +1,172 @@
+
+    /** With ONHEAP_TIERED policy. */
+    private final String ON_HEAP_TIERED_NAME = "onHeapTieredCfg";
+
+    /** {@inheritDoc} */
+    @Override protected int gridCount() {
+        return NODES_CNT;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String gridName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(gridName);
+
+        cfg.setIncludeProperties();
+        cfg.setMetricsUpdateFrequency(0);
+
+        if (!cacheInit) {
+            CacheConfiguration<Integer, Object> offHeapValuesCfg = cacheConfiguration(gridName);
+            offHeapValuesCfg.setName(OFF_HEAP_VALUE_NAME);
+            offHeapValuesCfg.setStatisticsEnabled(true);
+            offHeapValuesCfg.setMemoryMode(CacheMemoryMode.OFFHEAP_VALUES);
+            offHeapValuesCfg.setOffHeapMaxMemory(MAX_VALS_AMOUNT * VAL_SIZE);
+
+            CacheConfiguration<Integer, Object> onHeapTieredCfg = cacheConfiguration(gridName);
+            onHeapTieredCfg.setName(ON_HEAP_TIERED_NAME);
+            onHeapTieredCfg.setStatisticsEnabled(true);
+            onHeapTieredCfg.setMemoryMode(CacheMemoryMode.ONHEAP_TIERED);
+            onHeapTieredCfg.setOffHeapMaxMemory(MAX_VALS_AMOUNT * VAL_SIZE);
+
+            FifoEvictionPolicy plc = new FifoEvictionPolicy();
+            plc.setMaxMemorySize(MAX_VALS_AMOUNT * VAL_SIZE);
+            plc.setMaxSize(VAL_SIZE);
+
+            onHeapTieredCfg.setEvictionPolicy(plc);
+
+            cacheInit = true;
+
+            return cfg.setCacheConfiguration(offHeapValuesCfg, onHeapTieredCfg);
+        }
+
+        return cfg;
+    }
+
+    public void testAllocatedMemory() throws Exception {
+        final IgniteCache tiered = grid(0).getOrCreateCache(ON_HEAP_TIERED_NAME);
+        final IgniteCache values = grid(0).getOrCreateCache(OFF_HEAP_VALUE_NAME);
+
+        long prevTieredOffHeapSize = tiered.metrics().getOffHeapAllocatedSize();
+        long prevValuesOffHeapSize = values.metrics().getOffHeapAllocatedSize();
+
+        assertEquals(0, prevTieredOffHeapSize);
+        assertEquals(0, prevValuesOffHeapSize);
+
+        long prevClusterNonHeapMemoryUsed = grid(0).cluster().metrics().getNonHeapMemoryUsed();
+
+        fillCache(tiered, getTestTimeout());
+
+        assertTrue(tiered.metrics().getOffHeapAllocatedSize() > (MAX_VALS_AMOUNT - 5) * VAL_SIZE + prevTieredOffHeapSize);
+        assertEquals(prevValuesOffHeapSize, values.metrics().getOffHeapAllocatedSize());
+        // prevClusterNonHeapMemoryUsed = 69_277_368
+        // (MAX_VALS_AMOUNT - 5) * VAL_SIZE = 49_807_360
+        // tiered.metrics().getOffHeapAllocatedSize() = 51_012_531
+
+        assertTrue((MAX_VALS_AMOUNT - 5) * VAL_SIZE + prevClusterNonHeapMemoryUsed < grid(0).cluster().metrics().getNonHeapMemoryUsed());
```

Comment:
prevClusterNonHeapMemoryUsed should be 0 here. You can delete it from the calculation for the simplicity of understanding your code.